### PR TITLE
Feature: User & Post Management

### PR DESCRIPTION
**Pull request by cancelei**
This PR introduces functionalities for users and post-management. We've added routes, controllers, and views for displaying lists of all users, specific user details, all posts by a given user, and individual post details.

**Changes:**
**Routes:**

Root route now points to the list of all users.
New routes added for user details, user posts, and specific post details.

**Controllers:**
UsersController: Contains actions index and show.
PostsController: Contains actions index and show.

**Views:**
Created placeholder views for the above routes with basic HTML content.

**Database Seeding:**
Added seed data for users, posts, comments, and likes to facilitate local testing.

The views currently have placeholder content. Styling and final content will be added in subsequent PRs.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
-  # Defines the root path route ("/")
-  # root "articles#index"
+  root to: 'users#index' # this will map the "https://<your-domain>/" URL
+  
+  get 'users/:id', to: 'users#show', as: 'user'
+  get 'users/:user_id/posts', to: 'posts#index'
+  get 'users/:user_id/posts/:id', to: 'posts#show', as: 'user_post'
```
Comment (ShadyShawkat) [inline config/routes.rb]: - [ ] [OPTIONAL] You can use the following format to structure your routes which is cleaner.

```ruby
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```

**Review by ShadyShawkat — APPROVED** [review]
## Hi @cancelei, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 
- Your readme looks professional ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 1 above)
Comment (cancelei) [reply config/routes.rb]: @ShadyShawkat I did that, but now the root is displaying the rails welcome page. And localhost/users is displaying the root content.

Is it still okay to leave it like that? 

image

image

Comment (ShadyShawkat) [reply config/routes.rb]: @cancelei You can keep the root line in place and change the others to looks something like this:

```ruby 
root to: 'users#index'

resources :users, only: [:index, :show] do
  resources :posts, only: [:index, :show]
end
```

Comment (cancelei) [reply config/routes.rb]: It worked like a glove, thank you for the feedback! @ShadyShawkat

**Pull request merged**